File: app/src/main/java/nhn/calendarapp/ui/adapter/TaskItemAdapter.java
package nhn.calendarapp.ui.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.github.vipulasri.timelineview.TimelineView;

import java.util.ArrayList;

import nhn.calendarapp.R;
import nhn.calendarapp.model.Task;

/**
 * Created by nguyennguyen on 1/6/17.
 */

public class TaskItemAdapter extends RecyclerView.Adapter<TaskItemAdapter.ViewHolder> {

    private ArrayList<Task> taskList;

    public TaskItemAdapter(ArrayList<Task> taskList) {
        this.taskList = taskList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = View.inflate(parent.getContext(), R.layout.item_task, null);
        return new ViewHolder(view, viewType);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Task task = taskList.get(position);
        holder.tvTaskDesc.setText(task.getTaskDesc());
        holder.tvTaskName.setText(task.getTaskName());
        holder.tvTime.setText(task.getTaskTime());
    }

    @Override
    public int getItemCount() {
        return taskList.size();
    }

    @Override
    public int getItemViewType(int position) {
        return super.getItemViewType(position);
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        TimelineView timelineView;
        TextView tvTime, tvTaskName, tvTaskDesc;

        public ViewHolder(View itemView, int viewType) {
            super(itemView);
            timelineView = (TimelineView) itemView.findViewById(R.id.timeline_marker);
            timelineView.initLine(viewType);
            tvTime = (TextView) itemView.findViewById(R.id.tv_task_time);
            tvTaskName = (TextView) itemView.findViewById(R.id.task_name);
            tvTaskDesc = (TextView) itemView.findViewById(R.id.task_desc);
        }
    }

}

File: app/src/main/java/nhn/calendarapp/ui/adapter/TaskItemAdapter.kt
package nhn.calendarapp.ui.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import com.github.vipulasri.timelineview.TimelineView
import nhn.calendarapp.R
import nhn.calendarapp.data.Task
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by nguyennguyen on 1/6/17.
 */
class TaskItemAdapter constructor(var listener: onTaskClickListener) : RecyclerView.Adapter<TaskItemAdapter.ViewHolder>() {

    private var taskList: List<Task>? = null
    private lateinit var context: Context

    interface onTaskClickListener {
        fun onTaskClick(task: Task)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = View.inflate(parent.context, R.layout.item_task, null)
        context = parent.context
        return ViewHolder(view, viewType)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val sdf = SimpleDateFormat("hh:mm aaa", Locale.US)
        val task = taskList!![position]
        holder.tvTaskDesc.text = task.taskDesc
        holder.tvTaskName.text = task.taskName
        holder.tvTime.text = sdf.format(task.taskTime)
        holder.timelineView.setMarker(context.resources.getDrawable(R.drawable.ic_marker))
        holder.container.setOnClickListener { listener.onTaskClick(task) }
    }

    override fun getItemCount(): Int {
        return taskList!!.size
    }

    override fun getItemViewType(position: Int): Int {
        return super.getItemViewType(position)
    }

    fun setItem(tasks: List<Task>) {
        taskList = tasks
    }

    inner class ViewHolder(itemView: View, viewType: Int) : RecyclerView.ViewHolder(itemView) {

        var timelineView: TimelineView
        var tvTime: TextView
        var tvTaskName: TextView
        var tvTaskDesc: TextView
        var container: LinearLayout

        init {
            timelineView = itemView.findViewById(R.id.timeline_marker) as TimelineView
            timelineView.initLine(viewType)
            tvTime = itemView.findViewById(R.id.task_time) as TextView
            tvTaskName = itemView.findViewById(R.id.task_name) as TextView
            tvTaskDesc = itemView.findViewById(R.id.task_desc) as TextView
            container = itemView.findViewById(R.id.container) as LinearLayout
        }

    }

}
